pkg/api/api.go

```diff
@@ -240,6 +252,54 @@ func requestPostageBatchId(r *http.Request) ([]byte, error) {
 	return nil, errInvalidPostageBatch
 }
 
+type authKeyResponse struct {
+	Key string `json:"key"`
+}
+
+type roleRequest struct {
+	Role string `json:"role"`
+}
+
+func (s *server) authHandler(w http.ResponseWriter, r *http.Request) {
+	user, pass, ok := r.BasicAuth()
+
+	if !ok || !s.auth.Authorize(user, pass) {
+		s.logger.Debug("unauthorized")
+		s.logger.Error("unauthorized")
+		w.Header().Set("WWW-Authenticate", `Basic realm="Restricted"`)
+		jsonhttp.Unauthorized(w, "Unauthorized")
+		return
+	}
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		s.logger.Debugf("read auth request body: %v", err)
+		s.logger.Errorf("read auth request body: %v", err)
```

Comment:
Error messages should not reveal internal implementations, where the err passed here may do so. It should be descriptive to the user about the problem and debug log messages should provide more information about the internals for developers.

Reply:
I'm not sure how to achieve that, is there a relevant example somewhere?

Reply:
Log messages in other handlers as for example these ones https://github.com/ethersphere/bee/blob/master/pkg/api/bytes.go#L32 can reveal a pattern. Also the reasoning is explained in coding.md file in the root dir.